core/src/main/java/hudson/FilePath.java

```diff
@@ -683,6 +683,26 @@ public boolean equals(Object o) {
 
     }
 
+    /**
+     * Returns true iff the argument is an ancestor of the current file path.
+     * @param ancestor a file path
+     * @return true iff the argument is an ancestor of the current file path.
+     * @since TODO
+     */
+    public boolean isDescendantOf(FilePath ancestor) {
```

Comment:
Be very very careful and see discussion in [JENKINS-26838](https://issues.jenkins-ci.org/browse/JENKINS-26838).